docs/DeveloperGuide.md

```diff
@@ -36,3 +33,470 @@
+            case MEDIUM:
+                minDirection = 0;
+                maxDirection = 3;
+                break;
+            case HARD:
+                minDirection = 0;
+                maxDirection = 4;
+                break;
+        }
+    }
+
+    public static int getAiDirection() {
+        Random rand = new Random();
+        int direction = rand.nextInt(maxDirection - minDirection + 1) + minDirection;
+        assert direction >= minDirection && direction <= maxDirection : "Illegal AI direction generated!";
+        return direction;
+    }
+
+    public static int getMinDirection() {
+        return minDirection;
+    }
+
+    public static int getMaxDirection() {
+        return maxDirection;
+    }
+}
+```
+
+The constructor initializes the `minDirection` and `maxDirection` based on the `DifficultyLevel` parameter. The `getAiDirection()` method generates a random integer within the range of `minDirection` and `maxDirection` using the `Random` class from Java's `util` package.
+
+### Alternatives Considered
+
+An alternative design could be to have separate classes for different difficulty levels, each with its own implementation of the `getAiDirection()` method. However, this would lead to code duplication and make the codebase more difficult to maintain.
+
+## Player Class
+
+### Introduction
+
+The `Player` class utilizes Object-Oriented Programming principles to track and store a user's performance, implementing functionality that varies based on the player's skill level. As the foundation of our player management system, it encapsulates common attributes and behaviors essential for all player types.
+
+### Skill Level Subclasses
+
+Subclasses `BeginnerSkill`, `MediumSkill`, and `ExpertSkill` are derived from the `Player` base class, each tailored to represent different skill levels. These subclasses override specific methods to provide behaviors unique to each skill level, ensuring a dynamic and engaging gameplay experience.
+
+### Attributes
+
+- `name`: String - Identifies the player.
+- `matchCount`: int - Tracks the number of matches played.
+- `power`: int - Represents the player's power level.
+- `skill`: int - Indicates the player's skill level.
+
+### Methods
+
+- `printSelfInfo()`: Displays the player's basic information.
+- `printGoalBeforeShoot()`: Prepares the player for a shooting attempt.
+- `printGoalBeforeSave()`: Prepares the player for a saving attempt.
+- `printGoalAfterShoot(boolean goalScored, int direction)`: Shows the outcome of a shooting attempt.
+- `upgradePower(int level)`: Enhances the player's power based on the specified level.
+- `shootDirectionAdjust(int dir)`: Modifies the shooting direction for the player.
+- `shootDirectionFormula(int left, int right, int dir, int power)`: Computes the shooting direction.
+- `aiDirectionAdjust(int aiDir)`: Modifies the AI's direction to offer a challenge.
+- `rangeAdjust()`: Modifies the shooting range according to the game's difficulty.
```

Comment:
Clear, detailed explanation of the Player Class. Very comprehensive overview of the attributes and methods